### ISSUE TITLE
XML catalog rewrite not applied to schemaLocation

### ISSUE THREAD

**Issue by GitMensch** [redhat-developer/vscode-xml#534]
I don't know if this is a bug in this extension, in lemminx or in the definition...

Given a [registered catalog](https://github.com/redhat-developer/vscode-xml/blob/master/docs/Validation.md#xml-catalog-with-xsd) definition of

```xml
<catalog xmlns="urn:oasis:names:tc:entity:xmlns:xml:catalog">

  <rewriteURI
      uriStartString="http://foo.web/foo/"
      rewritePrefix="./" />

</catalog>
```

And an `xmlns="http://foo.web/foo/types.xsd"` the file used for validation is the local `types.xsd` as expected, no downloading is done.

But with a given definition of `xsi:schemaLocation="http://foo.web/foo http://foo.web/foo/types.xsd"` the XML Support output pane says:

~~~
[Info  - 11:59:08] Jun 28, 2021 11:59:08 org.eclipse.lemminx.uriresolver.CacheResourcesManager lambda$downloadResource$0()
Message: Downloading http://foo.web/foo/types.xsd to C:\Users\me\.lemminx\cache\http\foo.web\foo\types.xsd...
[Error - 11:59:08] Jun 28, 2021 11:59:08 org.eclipse.lemminx.uriresolver.CacheResourcesManager lambda$downloadResource$0()
Message: Error while downloading http://foo.web/foo/types.xsd to C:\Users\me\.lemminx\cache\http\foo.web\foo\types : [sun.security.provider.certpath.SunCertPathBuilderException] unable to find valid certification path to requested target
~~~

So... where is the bug? If it is in the definition: What would be the correct approach?

**Comment by angelozerr** [comment]
If I udensratnd correctly, validation is working but hyperlinkinside xsi:schemaLocation  is not working, is that?

**Comment by GitMensch** [comment]
No, in both cases the standard link behavior happens - the not existing URL is opened (opening the file after applying the catalog and in case of cached files [where the actual URL exists] to the cache would be very cool).

In the case of `xmlns` the URI rewrite via catalog works (the local file is used, not the not existing URL), in case of `xsi:schemaLocation` it doesn't (the language server tries to download the not existing URL) and errors.

**Comment by angelozerr** [comment]
I didn't know that we could use `XML catalog rewriteURI` with `xsi:schemaLocation`.  I don't know if it will be easy to fix it (is Xerces can support that?).
